subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,85 @@
+ project('speexdsp', 'c',
+  version : '1.2.0',
+  default_options: ['warning_level=2',
+                    'c_std=gnu99',
+                    'buildtype=release'],
+  license : 'bsd-like')
+
+lib_current = 1
+lib_revision = 0
+lib_age = 2
+lib_soversion = 1
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+add_project_arguments('-DHAVE_CONFIG_H', language: 'c')
+
+conf_data = configuration_data()
+
+# Check for SIMD instructions
+c_args = []
+if host_cpu_family.startswith('x86')
+  c_args += cc.get_supported_arguments(['-mfpmath=sse', '-msse2'])
+  conf_data.set10('USE_SSE', true)
+  conf_data.set10('USE_SSE2', true)
+
+elif host_cpu_family == 'arm'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-vfpv4'])
+  conf_data.set10('USE_NEON', true)
+
+elif host_cpu_family == 'aarch64'
+  c_args += cc.get_supported_arguments(['-mfpu=neon-fp-armv8'])
+  conf_data.set10('USE_NEON', true)
+endif
+
+foreach arg : c_args
+  if cc.has_argument(arg)
+    add_project_arguments(arg, language: 'c')
+  endif
+endforeach
+
+# Check for C99 variable-size arrays, or alloca() as fallback
+msg_use_alloca = false
+if cc.compiles('''static int x;
+                  char some_func (void) {
+                    char a[++x];
+                    a[sizeof a - 1] = 0;
+                    int N;
+                    return a[0];
+                  }''', name : 'C99 variable-size arrays')
+  conf_data.set('VAR_ARRAYS', 1)
+  msg_use_alloca = 'NO (using C99 variable-size arrays instead)'
```

Comment:
Meson is not happy if you try to change the type of a variable. `msg_use_alloca` is initialized to a bool and you re assigning a string to it here.

Reply:
Thanks @dnicolodi; this game from the existing `ogg` wrap.
I'm blowing all of this away - all C99 compilers must support vararrays (and the wrap requires  `c_std=c99` compliance), so these checks aren't needed.